### PR TITLE
Feature: Auditor-detector: public/external functions without `msg.sender` checks

### PR DESCRIPTION
**Pull request by alexroan**

### CODE REVIEW

File: aderyn_core/src/lib.rs [hunk 1]
```diff
@@ -52,21 +52,24 @@ where
 }
 
 fn run_auditor_mode(context: &WorkspaceContext) -> Result<(), Box<dyn Error>> {
-    let auditors_with_instances = get_auditor_detectors()
```
Comment (alexroan) [inline aderyn_core/src/lib.rs:55]: @TilakMaddy hopefully this makes it easier to build more auditor-mode detectors with ICF

Comment (TilakMaddy) [reply aderyn_core/src/lib.rs:55]: @alexroan I'm yet to look into it ... but on a side note I was wondering if we can bring sarif and auditor mode support to ICF *after* the merge. So basically if you have `--icf -o report.sarif` we will say sarif mode is not supported yet with `--icf`. ...and likewise if you have `--icf --auditor-mode` we will say `auditor-mode` is not supported yet.

File: reports/report.md [hunk 2]
```diff
@@ -957,25 +999,7 @@ The `ecrecover` function is susceptible to signature malleability. This means th
 
 
 
-## L-4: Deprecated OpenZeppelin functions should not be used
```
Comment (alexroan) [inline reports/report.md:960]: Why is this being deleted??? 🤔

File: reports/report.json [hunk 3]
```diff
@@ -843,7 +862,7 @@
   },
   "issue_count": {
     "high": 6,
-    "low": 23
+    "low": 22
```
Comment (alexroan) [inline reports/report.json:865]: Why??

File: reports/report.md (hunk 2 above)
Comment (alexroan) [reply reports/report.md:960]: OMG I've found the reason!

The ID `5463` (which is the ID of the source_unit for DeprecatedOZFunctions.sol) is duplicated! 0.8.19, 0.8.20 AND 0.8.25 all product the same ID for different ASTNodes, so they're being overwritten in the WorkspaceContext.

Holy crap @TilakMaddy , we definitely need the multiple WorkspaceContexts!

**Comment by alexroan** [comment]
On hold until ICF is ready.

File: reports/report.md (hunk 2 above)
Comment (TilakMaddy) [reply reports/report.md:960]: @alexroan Is this the only thing blocking?

**Comment by TilakMaddy** [comment]
Which one is the latest? `AttackSurface` or `CallsAndDelegates` ?

File: reports/report.md (hunk 2 above)
Comment (TilakMaddy) [reply reports/report.md:960]: @alexroan https://github.com/Cyfrin/aderyn/pull/459  <---- ICF doesn't do this. Look here !!

File: aderyn_core/src/audit/public_functions_no_sender_checks.rs [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    },
+};
+use std::{cmp::Ordering, collections::BTreeSet, error::Error};
+
+#[derive(Clone, Eq, PartialEq)]
+pub struct NoChecksInstance {
+    pub contract_name: String,
+    pub function_name: String,
+    pub function_kind: FunctionKind,
+}
```
Comment (TilakMaddy) [inline aderyn_core/src/audit/public_functions_no_sender_checks.rs:21]: Please change this to String

**Comment by TilakMaddy** [comment]
@alexroan  you can continue your work here https://github.com/Cyfrin/aderyn/pull/459

**Pull request closed without merging**